pkg/controller/deployment/deployment_controller.go

```diff
@@ -60,6 +60,12 @@ const (
 	maxRetries = 15
 )
 
+const (
+	// Structured Logging keys
+	klogKeyDeployment = "deployment"
```

Comment:
I would say that using raw key names instead of extracting const for logging is one example of justified breaking of "do not repeat yourself" rule. 

For this PR we can leave it as it is, but I would prefer avoiding creating pattern. Will try to add a section in migration guide about this.

Reply:
"do not repeat yourself" was not the reason I introduced the constants.

There is no way to validate that everyone uses the same keys for the same objects.
Someone can write "replicaSet", someone else can write ""replicaset" or make a typo.
The constants could be a way of ensuring the key consistency.

Anyway, I wasn't sure about that in the first place and wanted to get feedback on that.
Removed the constants for now.

Reply:
You mentioned important problem: typos in key names.

I don't think we should solve it via consts, but we should look into static analysis. 
Main problem with consts is that they don't scale. To standardize key naming we would need to have one global repository of consts for whole k/k repo. This is not a dependency that we want to maintain, also it makes it even harder for projects outside k/k.

For now we don't want to enforce closed set of keys to use, but we should prevent typos like you mentioned. Those simple cases should be easy to detect without introducing to much restrictions on key names. We could add simple `hack/verify` script that prevents merging code with typos.

After one recent review where most issues where 3-4 different ways to write `podSandboxID`. I think we should add it to Beta scope. Possibly we can extend static analysis piece we already planned.
/cc @ehashman